lib/Models/UrthecastCatalogGroup.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+/*global require*/
+var defineProperties = require('terriajs-cesium/Source/Core/defineProperties');
+var knockout = require('terriajs-cesium/Source/ThirdParty/knockout');
+
+var ModelError = require('./ModelError');
+var raiseErrorToUser = require('../Models/raiseErrorToUser');
+var CatalogGroup = require('./CatalogGroup');
+var inherit = require('../Core/inherit');
+var UrthecastCatalogItem = require('./UrthecastCatalogItem');
+var urthecastClient = require('urthecast');
+
+/**
+ * A {@link CatalogGroup} representing a collection of avalible Urthecast sensor platforms.
+ *
+ * @alias UrthecastCatalogGroup
+ * @constructor
+ * @extends CatalogGroup
+ *
+ * @param {Terria} terria The Terria instance.
+ */
+var UrthecastCatalogGroup = function(terria) {
+    CatalogGroup.call(this, terria);
+
+    /**
+     * Gets or sets a hash of names of blacklisted data layers.  A layer that appears in this hash
+     * will not be shown to the user.  In this hash, the keys should be the Title of the layers to blacklist,
+     * and the values should be "true".  This property is observable.
+     * @type {Object}
+     */
+    this.blacklist = undefined;
+
+    knockout.track(this, ['blacklist']);
+};
+
+inherit(CatalogGroup, UrthecastCatalogGroup);
+
+defineProperties(UrthecastCatalogGroup.prototype, {
+    /**
+     * Gets the type of data member represented by this instance.
+     * @memberOf UrthecastCatalogGroup.prototype
+     * @type {String}
+     */
+    type : {
+        get : function() {
+            return 'urthecast-group';
+        }
+    },
+
+    /**
+     * Gets a human-readable name for this type of data source, such as 'Urthcast Sensor Platforms'.
+     * @memberOf UrthecastCatalogGroup.prototype
+     * @type {String}
+     */
+    typeName : {
+        get : function() {
+            return 'Urthcast Sensor Platforms';
+        }
+    },
+});
+
+UrthecastCatalogGroup.prototype._load = function() {
+    var catalogGroup = this;
+    var apiKey = catalogGroup.terria.configParameters.urthecastApiKey;
+    var apiSecret = catalogGroup.terria.configParameters.urthecastApiSecret;
+    var urthecast = urthecastClient(apiKey, apiSecret);
+
+    urthecast.v1.satellite_tracker('/sensor_platforms', {}, function(err, res) {
```

Comment:
This asynchronous bit needs to be turned into a promise, and the promise needs to be returned, so that the group displays "Loading..." while the load is in progress instead of displaying "This group contains no items."  You can turn a normal callback into a promise by creating a deferred:

```
var deferred = when.defer();
```

And then returning its promise:

```
return deferred.promise;
```

Inside the callback, on success:

```
deferred.resolve()
```

And on failure:

```
deferred.reject();
```

If you pass a `ModelError` instance to reject, TerriaJS will display it to the user.